src/ServiceControl.Monitoring/Timings/LongValueOccurrenceHandler.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace ServiceControl.Monitoring.Timings
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus;
+    using Processing.RawData.NServiceBus.Metrics;
+    using Processing.Snapshot;
+
+    class TimingsReportHandler : IHandleMessages<LongValueOccurrences>
+    {
+        readonly TimingsStore processingTimeStore;
+        readonly CriticalTimeStore criticalTimeStore;
+
+        static string ProcessingTimeMessageType = "NServiceBus.Metrics.ProcessingTime";
+        static string CriticalTimeMessageType = "NServiceBus.Metrics.CriticalTime";
+
+        public TimingsReportHandler(ProcessingTimeStore processingTimeStore, CriticalTimeStore criticalTimeStore)
+        {
+            this.processingTimeStore = processingTimeStore;
+            this.criticalTimeStore = criticalTimeStore;
+        }
+
+        public Task Handle(LongValueOccurrences message, IMessageHandlerContext context)
+        {
+            var messageType = context.MessageHeaders[Headers.EnclosedMessageTypes];
```

Comment:
On the sending side there is no `LongValueOccurence` message at all. There is only a raw dispatcher that writes the bytes. The deserialization relies on two headers right now `ContentType` and `EclosedMessagesTypes`. First to identify the metrics message format and second to identify the name of the metric it holds.